java-checks/src/main/java/org/sonar/java/checks/FieldModifierCheck.java

```diff
@@ -31,10 +32,12 @@
 import org.sonar.plugins.java.api.tree.VariableTree;
 
 import java.util.List;
+import java.util.function.Predicate;
 
 @Rule(key = "S2039")
 public class FieldModifierCheck extends IssuableSubscriptionVisitor {
 
+  private static final Predicate<ModifiersTree> MODIFIERS_TREE_PREDICATE = new VisibleForTestingPredicate().negate();
```

Comment:
Could you write your predicate using a lambda instead?
Even if used at 2 different places, it would be cleaner from our point of view. And no need to introduce a new class with its unit test, as it can be unit tested using the check verification.